sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/Utility.java

```diff
@@ -419,55 +442,51 @@ public static ExtractKeyPhrasesResultCollection toExtractKeyPhrasesResultCollect
-//                        final HealthcareEntityRelationType relationType =
-//                            HealthcareEntityRelationType.fromString(healthcareRelation.getRelationType());
-
-//                        final Map<HealthcareEntity, HealthcareEntityRelationType> targetRelatedEntity =
-//                            entityRelationMap.getOrDefault(targetEntity, new HashMap<>());
-//                        targetRelatedEntity.putIfAbsent(sourceEntity, relationType);
-//                        entityRelationMap.putIfAbsent(targetEntity, targetRelatedEntity);
-//
-//                        if (healthcareRelation.isBidirectional()) {
-//                            final Map<HealthcareEntity, HealthcareEntityRelationType> sourceRelatedEntity =
-//                                entityRelationMap.getOrDefault(sourceEntity, new HashMap<>());
-//                            sourceRelatedEntity.putIfAbsent(targetEntity, relationType);
-//                            entityRelationMap.putIfAbsent(sourceEntity, sourceRelatedEntity);
-//                        }
-                    });
+                final List<HealthcareRelation> relations = documentEntities.getRelations();
+                if (!CoreUtils.isNullOrEmpty(relations)) {
+                    final List<HealthcareEntityRelation> entityRelations = relations.stream().map(
+                        healthcareRelation -> {
+                            final HealthcareEntityRelation entityRelation = new HealthcareEntityRelation();
+                            final RelationType relationType = healthcareRelation.getRelationType();
+                            if (relationType != null) {
+                                HealthcareEntityRelationPropertiesHelper.setRelationType(entityRelation,
+                                    HealthcareEntityRelationType.fromString(relationType.toString()));
+                            }
+
+                            final List<HealthcareRelationEntity> relationEntities = healthcareRelation.getEntities();
+                            List<HealthcareEntityRelationRole> relationRoles = relationEntities.stream().map(
+                                relationEntity -> {
+                                    final Integer healthcareEntityIndex =
+                                        getHealthcareEntityIndex(relationEntity.getRef());
+                                    final HealthcareEntityRelationRole relationRole =
+                                        new HealthcareEntityRelationRole();
+                                    HealthcareEntityRelationRolePropertiesHelper.setName(relationRole,
+                                        relationEntity.getRole());
+                                    HealthcareEntityRelationRolePropertiesHelper.setEntity(relationRole,
+                                        healthcareEntities.get(healthcareEntityIndex));
```

Comment:
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json#L1756

healthcare entity won't be null

Reply:
But what if the desired index in `get()` is not found?

Reply:
it will cause a runtime exception and this exception will pop up until the mono.error()
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeHealthcareEntityAsyncClient.java#L168

Reply:
Do we really need to fail the API or the property can be left unpopulated with logging the error message?